R/dbplyr.R

```diff
@@ -0,0 +1,282 @@
+#' @include DBI.R
+NULL
+
+temporaryErrorMessage <- paste0("You cannot use `temporary = TRUE`",
+  "when using a Pool object, since temporary tables are local to a ",
+  "connection, and there's no guarantee you'll get the same ",
+  "connection back next time. You must either create a permanent ",
+  "table, or checkout a connection from `pool` directly with ",
+  "`con <- poolCheckout(pool)`, and then release the connection ",
+  "back to the pool when you're finished (`poolReturn(con)`).")
+
+# --- These generics are set in dplyr (not database-specific)
+#' @export
+copy_to.Pool <- function(dest, df, name = deparse(substitute(df)),
+  overwrite = FALSE, ...) {
+    db_con <- poolCheckout(dest)
+    on.exit(poolReturn(db_con))
+    copy_to(db_con, df = df, name = name, overwrite = overwrite, ... = ...)
+}
+
+#' @export
+tbl.Pool <- function(src, from, ...) {
+  db_con <- poolCheckout(src)
+  on.exit(poolReturn(db_con))
+  tbl(db_con, from = from, ... = ...)
+}
+
+# --- These generics are set in dplyr (database-specific)
+#' @export
+db_analyze.Pool <- function(con, table, ...) {
+  db_con <- poolCheckout(con)
+  on.exit(poolReturn(db_con))
+  db_analyze(db_con, table = table, ... = ...)
+}
+
+#' @export
+db_begin.Pool <- function(con, ...) {
+  db_con <- poolCheckout(con)
+  on.exit(poolReturn(db_con))
+  db_begin(db_con, ... = ...)
+}
+
+#' @export
+db_commit.Pool <- function(con, ...) {
+  db_con <- poolCheckout(con)
+  on.exit(poolReturn(db_con))
+  db_commit(db_con, ... = ...)
+}
+
+#' @export
+db_compute.Pool <- function(con, table, sql, temporary = TRUE,
+  unique_indexes = list(), indexes = list(), ...) {
+    if (temporary) stop(temporaryErrorMessage)
+    db_con <- poolCheckout(con)
+    on.exit(poolReturn(db_con))
+    db_compute(db_con, table = table, sql = sql,
+      temporary = temporary, unique_indexes = unique_indexes,
+      indexes = indexes, ... = ...)
+}
+
+#' @export
+db_create_index.Pool <- function(con, table, columns, name = NULL,
+  unique = FALSE, ...) {
+    db_con <- poolCheckout(con)
+    on.exit(poolReturn(db_con))
+    db_create_index(db_con, table = table, columns = columns,
+      name = name, unique = unique, ... = ...)
+}
+
+#' @export
+db_create_indexes.Pool <- function(con, table, indexes = NULL,
+  unique = FALSE, ...) {
+    db_con <- poolCheckout(con)
+    on.exit(poolReturn(db_con))
+    db_create_indexes(db_con, table = table, indexes = indexes,
+      unique = unique, ... = ...)
+}
+
+#' @export
+db_create_table.Pool <- function(con, table, types, temporary = FALSE, ...) {
```

Comment:
I do wonder about changing the default value of `temporary` here. Obviously it's going to be annoying if it's `TRUE`, but it might be surprising if it's `FALSE`.